login/src/androidMain/kotlin/org/dhis2/mobile/login/authentication/data/repository/TwoFARepositoryImpl.kt

```diff
@@ -0,0 +1,30 @@
+package org.dhis2.mobile.login.authentication.data.repository
+
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import org.dhis2.mobile.login.authentication.domain.model.TwoFAStatus
+import org.dhis2.mobile.login.authentication.domain.repository.TwoFARepository
+import org.hisp.dhis.android.core.D2
+
+class TwoFARepositoryImpl(
+    private val d2: D2,
+) : TwoFARepository {
+    override fun getTwoFAStatus(): Flow<TwoFAStatus> =
```

Comment:
Why to return a flow? It has only one emission.
I would change these to return the TwoFAStatus. And don't forget to set all repository functions as suspend, as they will interact with the database.